data/tribes/buildings/productionsites/amazons/stone_workshop/init.lua

```diff
@@ -62,6 +62,8 @@ wl.Descriptions():new_productionsite_type {
             -- time total: 19 * 69 = 1311 sec
             "call=produce_shovel",
             "call=produce_spear_wooden",
+            -- firestones before felling_ax to make poor hamlet work without dismantle
+            "call=produce_firestones",
```

Comment:
Please don't break the pattern of one tool – one spear. (or at least keep the skip at the end as it is now)

As for bringing firestones so early for the sake of poor hamlet, I'm not sure it's worth it. Noone said poor hamlet would be easy. ;) My main problem was juggling workers around anyway, when I tried to test it. This change doesn't solve that, only helps a little if you made a mistake while doing it. But I'm not sure how it is with the other tribes.

(BTW Do you plan to finish your old PRs?)

Reply:
I've tested poor hamlet of Empire and Amazons so far (Crater/blue), and Empire has a similar issue with the stonemason (get enough marble from the quarry before you can send him to make marble columns). All in all, it took me a little over 4 hours game / 1h real time to get full tool and building material production with Empire, and just below 3h game / about 0.5h real time with Amazons. Keeping the quartz for firestones wasn't even the "hardest" part. Though with Empire you can slip up a little with tools: you can spare 2 of the 7 iron ores if you don't use a geologist and go for iron mining before marble (needs a lot of rocks).

As for destructing buildings for Amazons: the liana cutter is much worse than the woodcutter, because you need rope to make a jungle preserver.

Reply:
FYI: [This thread in the forum](https://www.widelands.org/forum/topic/4880/) explains some of the challenges of poor hamlet for empire. Especially the [post form](https://www.widelands.org/forum/post/33706/) explains the purpose of this starting condition.

Every tribe should have a similar challenge, imho.

Reply:
The stone workshop programs are arranged so that it always calls the program of a spear after a program of a tool. (adding firestones actually broke that at the end, because the spears are also iterated within this pattern and I didn't want to break their balance) Now you have 2 tools next to each other in the second step of this cycle.

-------

Meanwhile I tried Barbarians too. I wasn't aware that the wood hardener needs a lumberjack, but I was still able to prevent the production of the felling axe by dismantling first the lumberjack while I produced some hard wood, then I dismantled the hardener and rebuilt the lumberjack. It needed just over 3h game / about 0.5h real time, so comparable to Amazons.

Reply:
Atlanteans needed a little over 3h game / ~40min real without crystal mine, still well under 4h with it. They don't need juggling workers, and my use of logs was far from optimal, so I could probably do it in 3h gametime.

Reply:
Surprisingly, Frisians were the easiest. I was able to get all basic material production up in about 1.5h gametime. (adding a coalmine, which is not essential for them, only took a few more minutes). Actually this should probably be turned around a bit, and provide some clay and coal instead of bricks, so that the brick kiln must be built first, then well, claypit and charcoal burner afterwards from newly burned bricks.

-----

Conclusion: Amazons are in the middle, either with or without rearranging the tool programs. So this PR is not essential, but as Barbarians have a similar help, it's OK. But please correct the tool-spear-tool-spear cycle.